js/formDataToJson.js

```diff
@@ -91,6 +95,213 @@ async function copyToClipboard(event){
+			headers: {
+			  'Content-Type': 'application/json',
+			  'Authorization': `token ${token}`,
+			},
+			body: JSON.stringify({
+			  ref: `refs/heads/${NEW_BRANCH}`, // Name of the new branch
+			  sha: sha, // SHA of the base branch (main)
+			}),
+		});
+
+		const newBranchData = await newBranchResponse.json();
+
+		if ( newBranchResponse.ok )
+		{
+			console.log('New branch created successfully: ', newBranchData);
+			return true;
+		}
+		else
+		{
+			console.error('Error creating new branch: ', newBranchData);
+			alert("Failed to create branch on project! Error code: ", newBranchResponse.status, ". Please check API Key permissions and try again.")
+			return false;
+		}
+	}
+	else
+	{
+		console.error('Error fetching base branch info:', data);
+		return false;
+	}
+}
+
+
+async function addFileToBranch(projectURL, token, codeJSONObj)
+{
+	const {owner, repo} = getOrgAndRepoArgsGitHub(projectURL);
+	const FILE_PATH = 'code.json'
+	const createFileApiUrl = `https://api.github.com/repos/${owner}/${repo}/contents/${FILE_PATH}`;
+	const encodedContent = btoa(codeJSONObj);
+	console.log("Content: ", encodedContent);
+	console.log("Branch: ", NEW_BRANCH);
+
+	const response = await fetch(createFileApiUrl, 
+		{
+			method: 'PUT',
+			headers: {
+				'Accept': 'application/vnd.github+json',
+				'Authorization': 'Bearer '.concat(token),
+				'X-GitHub-Api-Version': "2022-11-28"
+			},
+			body: JSON.stringify({
+				message: "Add codejson to project",
+				committer: {
+					name: "codejson-generator form site",
+					email: "opensource@cms.hhs.gov"
+				},
+				content: encodedContent,
+				branch: NEW_BRANCH,
+			}),
+		}
+	);
+
+	const data = await response.json()
+
+	if ( response.ok )
+	{
+		console.log('File added successfully: ', data);
+		return true;
+	}
+	else
+	{
+		console.error('Error adding file: ', data);
+		alert("Failed to add file on project! Error code: ", response.status, ". Please check API Key permissions and try again.")
+		return false;
+	}
+}
+
+async function createPR(projectURL, token)
+{
+	const {owner, repo} = getOrgAndRepoArgsGitHub(projectURL);
+	const createPrApiUrl = `https://api.github.com/repos/${owner}/${repo}/pulls`;
+	const response = await fetch(createPrApiUrl, 
+		{
+			method: 'POST',
+			headers: {
+				'Content-Type': 'application/json',
+				'Authorization': 'token '.concat(token),
+				'X-GitHub-Api-Version': "2022-11-28"
+			},
+			body: JSON.stringify({
+				title: "Add code.json to Project",
+				body: "Add generated code.json file to project. Code.json was generated via codejson-generator form site.",
+				head: NEW_BRANCH,
+				base: 'main',
+
+			}),
+		}
+	);
+
+	const data = await response.json();
+
+	if (response.ok)
+	{
+		console.log('Pull request created successfully: ', data);
+		return true;
+	}
+	else
+	{
+		console.error("Error creating PR!: ", data);
+		alert("Failed to create PR on project! Error code: ", response.status, ". Please check API Key permissions and try again.")
```

Comment:
Did some error handling testing and error code is not showing. I think it's cause the commas should be `+` or consider using template literals

<img width="694" alt="Screenshot 2025-04-21 at 2 56 56 PM" src="https://github.com/user-attachments/assets/add87328-b793-450d-aa5d-b92339baffbd" />